### PR TITLE
Add paging for space key and ability to dfs children pages

### PR DESCRIPTION
**Pull request by nikunjy**
1. `space_key` will now page through the confluence API in a limited way as opposed to requesting a very large limit 
2. Ability to DFS children. I believe this is going to be used more often than not.

If you like the children option I can do it for the `label` option as well.

### CODE REVIEW

**Comment by nikunjy** [comment]
if interested in merging please review @emptycrown  / @jerryjliu  

It was useful to me when I was using it.

File: loader_hub/confluence/base.py [hunk 1]
```diff
@@ -92,13 +106,26 @@ def load_data(self, space_key: Optional[str] = None, page_ids: Optional[List[str
...
+            if include_children:
+                page_ids = self._dfs_page(self.confluence, page_ids[0])
             for page_id in page_ids:
                 page = self.confluence.get_page_by_id(page_id=page_id, expand='body.storage.value')
                 doc = self.process_page(page, include_attachments, text_maker)
                 docs.append(doc)
 
         return docs
 
+    def _dfs_page(self, raw_confluence, page_id):
```
Comment (nikunjy) [inline loader_hub/confluence/base.py:120]: I can combine this and the getting the page itself as well but keeping it this way for now.

**Comment by jerryjliu** [comment]
thanks! will review tomorrow

File: loader_hub/confluence/base.py [hunk 2]
```diff
@@ -34,7 +34,7 @@ def __init__(self, base_url: str = None, oauth2: Optional[Dict] = None) -> None:
             from atlassian import Confluence
         except ImportError:
             raise ImportError("`atlassian` package not found, please run `pip install atlassian-python-api`")
-
+        self.confluence: Confluence = None
```
Comment (jerryjliu) [inline loader_hub/confluence/base.py:37]: you don't need this line (it's set in the if/else below)

**Review by jerryjliu — APPROVED** [review]
amazing!

**Comment by nikunjy** [comment]
Thanks for merging, was traveling so couldn't do it.

**Pull request merged**